pkg/env/readers.go

```diff
@@ -116,7 +128,7 @@ func testReader(fallback cfg.Reader, overrider *testOverrides) cfg.Reader {
 // use the fName across two tests running in parallel. This will cause the
 // function to potentially panic.
 //
-// TODO[DT-3185]: Related work item to make the safety of this function better
+// Deprecated: Please use `FakeTestConfigHandler`
```

Comment:
You should replace the call on line 135 with a call to the new `addHandler` function - and delete the old `add` function.

This function can now have the `panic` statement, and you can have the note about this being bad & deprecated.